app/models/piece.rb

```diff
@@ -47,44 +47,25 @@ def capture_position(destination, destination_x, destination_y)
   def diagonal_obstruction?(destination_x, destination_y, x_difference, y_difference)
     return false unless x_difference.abs == y_difference.abs && x_difference != 0
     y_direction = y_difference <=> 0
-    if x_difference < 0
-      ((x_position + 1)...destination_x).each_with_index do |x, i|
-        y = y_position + ((i + 1) * -y_direction)
-        return true if game.pieces.exists?(x_position: x, y_position: y)
-      end
-    else
-      ((destination_x + 1)...x_position).each.with_index do |x, i|
-        y = destination_y + ((i + 1) * y_direction)
-        return true if game.pieces.exists?(x_position: x, y_position: y)
-      end
+    (x_difference < 0 ? ((x_position + 1)...destination_x) : ((destination_x + 1)...x_position)).each_with_index do |x, i|
```

Comment:
@ryanveroniwooff this and the other two obstruction methods you refactored here are awesome